File: finalAssignmentBackend/src/main/java/com/tutict/finalassignmentbackend/entity/AppealManagement.java
package com.tutict.finalassignmentbackend.entity;

import com.baomidou.mybatisplus.annotation.*;
import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Data;

import java.io.Serial;
import java.io.Serializable;
import java.time.LocalDateTime;

@Data
@TableName("appeal_management")
public class AppealManagement implements Serializable {
    @Serial
    private static final long serialVersionUID = 1L;


    @TableId(value = "appeal_id", type = IdType.AUTO)
    private Integer appealId;

    @TableField("offense_id")
    @JsonProperty("offenseId") // Map both field names
    private Integer offenseId;

    @TableField("appellant_name")
    @JsonProperty("appellant_name")
    private String appellantName;

    @TableField("id_card_number")
    @JsonProperty("id_card_number")
    private String idCardNumber;

    @TableField("contact_number")
    @JsonProperty("contact_number")
    private String contactNumber;

    @TableField("appeal_reason")
    @JsonProperty("appeal_reason")
    private String appealReason;

    @TableField("appeal_time")
    @JsonProperty("appeal_time")
    private LocalDateTime appealTime;

    @TableField("process_status")
    @JsonProperty("process_status")
    private String processStatus;

    @TableField("process_result")
    @JsonProperty("process_result")
    private String processResult;

    /**
     * 软删除时间戳
     */
    @TableField("deleted_at")
    @TableLogic(value = "null", delval = "now()")
    private LocalDateTime deletedAt;
}

File: final_asssignment_backend_go/project/internal/domain/appeal_management.go
package domain

import (
	"time"

	"gorm.io/gorm"
)

// AppealManagement 表示 appeal_management 表的实体
type AppealManagement struct {
	AppealID      int            `gorm:"column:appeal_id;primaryKey;autoIncrement" json:"appeal_id"`
	OffenseID     int            `gorm:"column:offense_id" json:"offenseId"`
	AppellantName string         `gorm:"column:appellant_name" json:"appellant_name"`
	IDCardNumber  string         `gorm:"column:id_card_number" json:"id_card_number"`
	ContactNumber string         `gorm:"column:contact_number" json:"contact_number"`
	AppealReason  string         `gorm:"column:appeal_reason" json:"appeal_reason"`
	AppealTime    time.Time      `gorm:"column:appeal_time" json:"appeal_time"`
	ProcessStatus string         `gorm:"column:process_status" json:"process_status"`
	ProcessResult string         `gorm:"column:process_result" json:"process_result"`
	DeletedAt     gorm.DeletedAt `gorm:"index"` // 软删除字段（可选，GORM 默认支持）
}

// TableName 指定数据库表名
func (AppealManagement) TableName() string {
	return "appeal_management"
}
